File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        current_sum = 0
        ans = float('inf')

        for right in range(len(nums)):
            current_sum += nums[right]
            while current_sum >= target:
                ans = min(ans, right - left + 1)
                current_sum -= nums[left]
                left += 1

        return ans if ans != float('inf') else 0
        

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    longestOnes(nums, k) {
        let left = 0
        let maxLength = 0

        for(let right = 0; right < nums.length; right++){
            if(nums[right] === 0){
                k-=1
            }

            while(k < 0){
                if (nums[left] === 0){
                    k+=1
                } 
                left+=1
            }

            maxLength = Math.max(maxLength, right - left + 1)
        }

        return maxLength;
    }
}
